File: Facade/homeTheatre.js
class dvdPlayer{
    turnOn(){
        console.log("dvd player turned on");
        return true;
    }
    turnOff(){
        console.log("dvd player turned off");
    }
}

class projectorPlayer{
    turnOn(){
        console.log("projector turned on");
        return true;
    }
    turnOff(){
        console.log("projector turned off");
    }
}

class soundSystemPlayer{
    turnOn(){
        console.log("sound system turned on");
        return true;
    }
    turnOff(){
        console.log("sound system turned off");
    }
}

class facadeTheatre{
    constructor(){
        this.dvd = new dvdPlayer();
        this.projector = new projectorPlayer();
        this.soundSystem = new soundSystemPlayer();
    }
    turnMovieOn(){
        if(!this.dvd.turnOn()){
            throw new Error("turn on dvd");
        }
        if(!this.projector.turnOn()){
            throw new Error("turn on projector");
        }
        this.soundSystem.turnOn();
    }

    turnMovieOff(){
        this.dvd.turnOff();
        this.projector.turnOff();
        this.soundSystem.turnOff();
    }

}

const client = new facadeTheatre();
client.turnMovieOn();
client.turnMovieOff();

File: Facade/homeTheatre.java
package Facade;
class dvdPlayer{ //Sub systems
    public void turnOn(){
        System.out.println("dvd player turned on");
    }
    public void turnOff(){
        System.out.println("dvd player turned off");
    }
}
class projector{
    public void turnOn(){
        System.out.println("projector turned on");
    }
    public void turnOff(){
        System.out.println("projector turned off");
    }
}
class musicPlayer{
    public void turnOn(){
        System.out.println("music player turned on");
    }
    public void turnOff(){
        System.out.println("music player turned off");
    }
}
class Facade{ //facade
    private dvdPlayer dp;
    private projector p;
    private musicPlayer mp;
    Facade(){
        this.dp = new dvdPlayer();
        this.p = new projector();
        this.mp = new musicPlayer();
    }
    public void turnOn(){
        dp.turnOn();
        p.turnOn();
        mp.turnOn();
    }
    public void turnOff(){
        dp.turnOff();
        p.turnOff();
        mp.turnOff();
    }
}
public class homeTheatre {
    public static void main(String[] args) {
        Facade client =new Facade(); //Client
        client.turnOn();
        client.turnOff();
    }
}
